semana10/futurex/src/containers/ApplicationForm/index.js

```diff
@@ -0,0 +1,356 @@
+import React, { Component } from "react";
+import { connect } from "react-redux";
+import { createCandidate } from '../../action'
+import { push, replace, goBack } from "connected-react-router";
+import { routes } from "../Router";
+
+const formApplication = [
+    {
+      name: "name",
+      type: "text",
+      label: "Nome:",
+      required: false,
+      pattern: "[A-Za-z]{3,}"
+    },
+    {
+      name: "age",
+      type: "number",
+      label: "Idade: ",
+      required: false,
+      minValue: 18,
+    },
+    {
+      name: "applicationText",
+      type: "text",
+      label: "Sou bom candidato por: ",
+      required: false,
+      minValue: 30
+    },
+    {
+      name: "profession",
+      type: "text",
+      label: "Profissão",
+      required: false,
+      minValue: 10
+    },
+  ];
+
+class ApplicationForm extends Component {
+  constructor(props){
+    super(props);
+      this.state = {
+        form:{}
+      }
+    }
+
+
+handleInputChange = event => {
+  const { name, value } = event.target;
+  this.setState({ form: { ...this.state.form, [name]: value } });
+};
+  
+handleCreateCandidate = (event) => {
+  event.preventDefault();
+  const {name, age, applicationText, profession, country, tripId} = this.state.form
+  this.props.createCandidate(name, age, applicationText, profession, country, tripId)
```

Comment:
Aqui como ideia de melhoria, lembre-se que um dos smells que vimos de código era uma função que recebia muitos parâmetros. Aqui seria mais fácil passar o this.state.form inteiro como objeto, e só extrair os dados de dentro na action.